iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -303,6 +308,58 @@ static LogicalResult convertSubTensorOp(
       allocationDynamicSizes, bvm);
 }
 
+/// Converts a `subtensor_insert` operation to buffers by
+/// - Allocating a buffer for the result (if needed), and copying the
+///   destination value into this tensor.
+/// - Copying the source values into a subview of the result.
+static LogicalResult convertSubTensorInsertOp(
+    OpBuilder &b, WorkgroupMemoryAllocationFn allocationFn,
+    SubTensorInsertOp op, BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = op.getLoc();
+  Value dest = op.dest();
+  Value inputBuffer = bvm.lookup(dest);
+  auto allocationDynamicSizes = llvm::to_vector<4>(llvm::map_range(
+      llvm::seq<unsigned>(0,
+                          inputBuffer.getType().cast<ShapedType>().getRank()),
+      [&](int64_t dim) {
+        return b.createOrFold<DimOp>(loc, inputBuffer, dim);
+      }));
+  if (failed(createAliasingBufferOrAllocationForResult(
+          b, loc, allocationFn, dest, inputBuffer, op.getResult(),
+          allocationDynamicSizes, bvm))) {
+    return success();
+  }
+
+  Value source = op.source();
+  Value outputBuffer = bvm.lookup(op.result());
+  Value sourceBuffer = bvm.lookup(source);
+  auto subViewResultType =
+      SubViewOp::inferResultType(outputBuffer.getType().cast<MemRefType>(),
+                                 extractFromI64ArrayAttr(op.static_offsets()),
+                                 extractFromI64ArrayAttr(op.static_sizes()),
+                                 extractFromI64ArrayAttr(op.static_strides()));
+  auto subViewOp =
+      b.create<SubViewOp>(loc, subViewResultType, outputBuffer, op.offsets(),
+                          op.sizes(), op.strides(), op.static_offsets(),
+                          op.static_sizes(), op.static_strides());
```

Comment:
Can this be:

```
auto subViewOp = b.create<SubViewOp>(loc, outputBuffer, op.getMixedOffsets(), op.getMixedSizes(), op.getMixedStrides());
```

There is a builder in SubViewOp:

```
    // Build a SubViewOp with mixed static and dynamic entries and custom
    // result type. If the type passed is nullptr, it is inferred.
    OpBuilderDAG<(ins "Value":$source, "ArrayRef<OpFoldResult>":$offsets,
      "ArrayRef<OpFoldResult>":$sizes, "ArrayRef<OpFoldResult>":$strides,
      CArg<"ArrayRef<NamedAttribute>", "{}">:$attrs)>,
```

The implementation is here:https://github.com/llvm/llvm-project/blob/c907681b077c6acf95c3ddca6611639c29559e40/mlir/lib/Dialect/StandardOps/IR/Ops.cpp#L2991-L2997

It will infer the type and you can pass mixed offsets/sizes/strides.

Reply:
Thanks. That works well!